bundles/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/internal/handler/EnergeniePWMHandler.java

```diff
@@ -0,0 +1,201 @@
+ *
+ * @author Hans-Jörg Merk - Initial contribution
+ */
+
+@NonNullByDefault
+public class EnergeniePWMHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EnergeniePWMHandler.class);
+
+    private String host = "";
+    private String password = "";
+
+    /** the timeout to use for connecting to a given host (defaults to 5000 milliseconds) */
+    private int timeout = 6000;
+
+    /**
+     * The default refresh interval in Seconds.
+     */
+    private int DEFAULT_REFRESH_INTERVAL = 60;
+    @Nullable
+    private ScheduledFuture<?> refreshJob;
+    private Runnable refreshRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            getState();
+        }
+    };
+
+    public EnergeniePWMHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // PWM devices don't support any commands, just value reading
+    }
+
+    @Override
+    public void initialize() {
+        Configuration config = getConfig();
+
+        if ((config.get("host") != null) && (config.get("password") != null)) {
+            host = (String) config.get("host");
+            password = (String) config.get("password");
+            logger.debug("Initializing EnergeniePWMHandler for Host '{}'", host);
+            updateStatus(ThingStatus.ONLINE);
+            onUpdate();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Can not access device , IP-Address or password not set");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("EnergeniePWMHandler disposed.");
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    public void getState() {
+        String url = "http://" + host + "/login.html";
+        String urlParameters = "pw=" + password;
+        InputStream urlContent = new ByteArrayInputStream(urlParameters.getBytes(Charset.forName("UTF-8")));
+        String loginResponseString = null;
+
+        try {
+            logger.trace("sendlogin to {} with password {}", host, password);
+            logger.trace("sending 'POST' request to URL : {}", url);
+            loginResponseString = HttpUtil.executeUrl("POST", url, urlContent, "TEXT/PLAIN", timeout);
+
+            if (loginResponseString != null) {
+                readState(loginResponseString, "voltage");
+                readState(loginResponseString, "current");
+                readState(loginResponseString, "power");
+                readState(loginResponseString, "energy");
+                try {
+                    HttpUtil.executeUrl("POST", url, timeout);
+                    logger.trace("logout from ip {}", host);
+                } catch (Exception e) {

```

Comment:
```suggestion
                } catch (IOException e) {
```
All all other places this is done.

Reply:
Done